fault.h

```diff
@@ -89,6 +128,127 @@ struct kibosh_fault_read_delay {
     double fraction;
 };
 
+/**
+ * The class for Kibosh faults that make files unwritable.
+ */
+struct kibosh_fault_unwritable {
+    /**
+     * The base class members.
+     */
+    struct kibosh_fault_base base;
+
+    /**
+     * The path prefix, starts with '/'.
+     */
+    char *prefix;
+
+    /**
+     * The type of file to be made unwritable.
+     */
+    char *file_type;
+
+    /**
+     * The error code to return from write faults.
+     */
+    int code;
+};
+
+/**
+ * The class for Kibosh faults that lead to corrupted data when reading.
+ */
+struct kibosh_fault_read_corrupt {
+    /**
+     * The base class members.
+     */
+    struct kibosh_fault_base base;
+
+    /**
+     * The path prefix, starts with '/'.
+     */
+    char *prefix;
+
+    /**
+     * The type of file to be corrupted.
+     */
+    char *file_type;
+
+    /**
+     * Mode of byte corruption.
+     * 1000 -> zeros (default)
+     * 1001 -> random values
+     * 1100 -> sequential zeros
+     * 1101 -> sequential random values
+     * 1200 -> silently drop a fraction of bytes at the end of buffer
+     */
+    int mode;
+
+    /**
+     * Number of corruption fault injected before switching to unwritable fault.
+     * Less than 0 means never switch.
+     */
+    int count;
+
+    /**
+     * The fraction of bytes to be corrupted.
+     */
+    double fraction;
+
+    /**
+     * Whether to store corrupted data to log.
+     * 0 means do not store data.
+     * Greater than 0 means store data to log.
+     */
+    int store_data;
```

Comment:
What does it mean to have this set for a read operation?

Reply:
This flag is added to prevent unnecessary logs. In corruption faults, the corrupted data would be written to the logs for future references. User can set this flag to 0 to not store the data (can be large). Setting the flag to greater than 0 means writing corrupted data to Kibosh logs.